python/ray/_private/node.py

```diff
@@ -176,12 +169,14 @@ def __init__(
             self._init_gcs_client()
 
         # Register the temp dir.
-        if head:
-            # date including microsecond
-            date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
-            self._session_name = f"session_{date_str}_{os.getpid()}"
-        else:
-            if ray_params.session_name is None:
+        self._session_name = ray_params.session_name
+
+        if self._session_name is None:
+            if head:
+                # date including microsecond
+                date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
+                self._session_name = f"session_{date_str}_{os.getpid()}"
+            else:
                 assert not self._default_worker
                 session_name = ray._private.utils.internal_kv_get_with_retry(
```

Comment:
（This not a broker) .  the behavior of worker node will be changed if  worker node also set `RAY_external_storage_namespace`.  
origin:
worker node's session_name always fetch  from kv storage.

Now:
If worker node set `RAY_external_storage_namespace`. worker node's session_name will use `RAY_external_storage_namespace` first.

If the RAY_external_storage_namespace is set improperly by the user, it can potentially cause issues with the worker nodes.